src/execution_engine.rs

```diff
@@ -305,21 +305,47 @@ impl ExecutionEngine {
 // want owned modules to drop.
 impl Drop for ExecutionEngine {
     fn drop(&mut self) {
-        forget(self.target_data.take().expect("TargetData should always exist until Drop"));
+        forget(
+            self.target_data
+                .take()
+                .expect("TargetData should always exist until Drop"),
+        );
+    }
+}
+
+impl Clone for ExecutionEngine {
```

Comment:
Should the EE really get a publicly accessible clone method? `Clone` generally means reallocating a new copy, but here we're just making a new reference to the same EE